[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104427
* #104418
* __->__ #104409
* #104408
* #104407
* #104346

`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

Commit:
[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

voznesenskym committed:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

fegin reviewed (approved):
Nice, I got confused by `_get_fully_sharded_module_to_states` when it was introduced to the code.

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]

Commit:
Update on "[FSDP][4/N] Remove `_get_fully_sharded_module_to_states`"


`_get_fully_sharded_module_to_states()` was used to emulate auto wrapping without actually calling `fully_shard`. Since we committed to unifying (see previous PR), we can remove this function and its helpers/tests.

[ghstack-poisoned]
